src/votekit/models.py

```diff
@@ -1,34 +1,57 @@
-from pydantic import BaseModel
-from typing import Optional
+from abc import ABC, abstractmethod
+from typing import Union, Any
+from .profile import PreferenceProfile
+from .election_state import ElectionState
+from .ballot_generator import BallotGenerator
 
 
-# Example of immutable data model for results
-class Outcome(BaseModel):
+class Simulation(ABC):
     """
-    elected (a set of Candidate): candidates who pass a certain threshold to win an election
-    eliminated (a set of Candidate): candidates who were eliminated (lost in the election)
-    remaining (a set of Candidate): candidates who are still in the running
-    rankings (a list of a set of Candidate): ranking of candidates with sets representing ties
+    Base class for model complex elections or statewide simulations as
+    done in MGGG's RCV research.
+
+    *Attributes*
+
+        Ballots: PreferenceProfile or dictionary of ballot generators
     """
 
-    # TODO: replace these with list/set[Candidate] once candidate is well-defined?
-    remaining: set[str]
-    elected: set[str] = set()
-    eliminated: set[str] = set()
-    votes: Optional[dict] = None
-    # TODO: re-add this
-    # rankings: list[set[str]]
-
-    class Config:
-        allow_mutation = False
-
-    def add_winners_and_losers(self, winners: set[str], losers: set[str]) -> "Outcome":
-        # example method, feel free to delete if not useful
-        if not winners.issubset(self.remaining) or not losers.issubset(self.remaining):
-            missing = (winners - self.remaining) | (losers - self.remaining)
-            raise ValueError(f"Cannot promote winners, {missing} not in remaining")
-        return Outcome(
-            remaining=self.remaining - winners - losers,
-            elected=self.elected | winners,
-            eliminated=self.eliminated | losers,
-        )
+    def __init__(self, ballots: Union[PreferenceProfile, dict, None] = None):
+        if ballots:
```

Comment:
it might be worth assigning self.ballots even if not true, so that the attribute at least exists (a missing attribute error down the line would likely be surprising)